modules/navier_stokes/test/tests/finite_volume/pwcns/channel-flow/2d-transient-gas.i

```diff
@@ -0,0 +1,372 @@
+    k = ${k}
+    porosity = porosity
+  []
+  [energy_convection]
+    type = PINSFVEnergyAmbientConvection
+    variable = T_fluid
+    is_solid = false
+    T_fluid = T_fluid
+    T_solid = T_solid
+    h_solid_fluid = 'h_cv'
+  []
+
+  [solid_energy_time]
+    type = PINSFVEnergyTimeDerivative
+    variable = T_solid
+    cp = ${cp_s}
+    rho = ${rho_s}
+    is_solid = true
+    porosity = porosity
+  []
+  [solid_energy_diffusion]
+    type = FVDiffusion
+    variable = T_solid
+    coeff = ${k_s}
+  []
+  [solid_energy_convection]
+    type = PINSFVEnergyAmbientConvection
+    variable = T_solid
+    is_solid = true
+    T_fluid = T_fluid
+    T_solid = T_solid
+    h_solid_fluid = 'h_cv'
+  []
+[]
+
+[FVBCs]
+  [inlet-u]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = superficial_vel_x
+    function = ${u_inlet}
+  []
+  [inlet-v]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = superficial_vel_y
+    function = 0
+  []
+  [inlet-T]
+    type = FVDirichletBC
+    variable = T_fluid
+    value = ${T_inlet}
+    boundary = 'left'
+  []
+
+  [no-slip-u]
+    type = INSFVNoSlipWallBC
+    boundary = 'top'
+    variable = superficial_vel_x
+    function = 0
+  []
+  [no-slip-v]
+    type = INSFVNoSlipWallBC
+    boundary = 'top'
+    variable = superficial_vel_y
+    function = 0
+  []
+  [heated-side]
+    type = FVDirichletBC
+    boundary = 'top'
+    variable = 'T_solid'
+    value = ${top_side_temperature}
+  []
+
+  [symmetry-u]
+    type = PINSFVSymmetryVelocityBC
+    boundary = 'bottom'
+    variable = superficial_vel_x
+    u = superficial_vel_x
+    v = superficial_vel_y
+    mu = ${mu}
+    momentum_component = 'x'
+  []
+  [symmetry-v]
+    type = PINSFVSymmetryVelocityBC
+    boundary = 'bottom'
+    variable = superficial_vel_y
+    u = superficial_vel_x
+    v = superficial_vel_y
+    mu = ${mu}
+    momentum_component = 'y'
+  []
+  [symmetry-p]
+    type = INSFVSymmetryPressureBC
+    boundary = 'bottom'
+    variable = pressure
+  []
+
+  [outlet-p]
+    type = INSFVOutletPressureBC
+    boundary = 'right'
+    variable = pressure
+    function = ${p_outlet}
+  []
+[]
+
+[FluidProperties]
+  [fp]
+    type = IdealGasFluidProperties
+    gamma = 1.4
+  []
+[]
+
+[FunctorMaterials]
+  [fluid_props_to_mat_props]
+    type = GeneralFunctorFluidProps
+    fp = fp
+    pressure = 'pressure'
+    T_fluid = 'T_fluid'
+    speed = 'velocity_norm'
+
+    # To initialize with a high viscosity
+    mu_rampdown = 'mu_rampdown'
+
+    # For porous flow
+    characteristic_length = 1
+    porosity = 'porosity'
+  []
+  [ins_fv]
+    type = INSFVEnthalpyFunctorMaterial
+    rho = ${rho}
+    temperature = 'T_fluid'
+  []
+  [constants]
+    type = ADGenericFunctorMaterial
+    prop_names = 'h_cv'
+    prop_values = '${h_fs}'
+  []
+[]
+
+[Functions]
+  [mu_rampdown]
+    type = PiecewiseLinear
+    x = '1 2 3 4'
+    y = '1e3 1e2 1e1 1'
+  []
+[]
+
+[AuxKernels]
+  [speed]
```

Comment:
There's also the `PINSFVSpeedFunctorMaterial`

Reply:
changed

Reply:
changed